bash/cloner.sh

```diff
@@ -25,26 +38,55 @@ make_folder () {
   echo
 }
 
-clone_repos () {
-  curl -s https://api.github.com/users/$account_name/repos | jq -c '.[]' |
+check () {
+  if [ -n "$1" ]
+    then # non-null/non-zero string check ... aka exists!
+      echo "success"
+      return 0; # return something similar to an exit code.
+    else
+      # echo "error"
+      return 1; # return something similar to an exit code.
+  fi
+}
 
-  while read i;
-  do
-    val=$(jq -r '.clone_url' <<< "$i");
-    (git clone -q $val) &> /dev/null & 
-  done;
+get_repos_by_page () {
+  local page=$1;
+  # reasons the curl request to the api can fail:
+  # 1. api request limit exceeded, clone_url property will not exist.
+  # 2. repository does not exist, clone url property will not exist.
+  local page_contents=$(curl https://api.github.com/users/$account_name/repos?page=$page | jq -c '.[]' 2>/dev/null | jq -r '.clone_url' 2>/dev/null);
 
-  echo -n loading;
+  # echo $page_contents;
+  check $page_contents &&
+  {
+    for url in ${page_contents}; do
+      echo "$url";
+
+      # git clone does not have a rate limiting that I am aware of.
+      (git clone -q $url) &> /dev/null &
+    done
+    echo ;
 
+    # recurse and go to the next page number until no repos come back. This might need a delay or sleep for large accounts.
+    get_repos_by_page $((page+1));
+  } || return 0;
```

Comment:
Something I just realized. This solution will recurse until an empty page is found. This means the last request will be unused. The alternative to prevent this "useless" request is to get the user info first and find the number of public repos to determine the number of pages. This of course costs one request.

Summary: Both approaches cost the same number of API requests.